GithubProfileWiki/Networking/APIService.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  APIService.swift
+//  GithubProfileWiki
+//
+//  Created by ilter on 23.01.2022.
+//
+
+import Foundation
+
+class APIService<T: APIHandler> {
+    let apiRequest: T
+    let urlSession: URLSession
+    
+    init(apiRequest: T, urlSession: URLSession = .shared) {
+        self.apiRequest = apiRequest
+        self.urlSession = urlSession
+    }
+    
+    func submitRequest(requestData: T.RequestDataType, completionHandler: @escaping (T.ResponseDataType?, Error?) -> ()) {
```

Comment:
you can work with async-await or you can write another network layer with that to adapt ot

Reply:
I will create another one, I just made it like this to learn all hands URLSession, Task, and completions from scratch.